conclave-common/src/test/kotlin/com/r3/conclave/common/EnclaveConstraintTest.kt

```diff
@@ -278,6 +280,54 @@ class EnclaveConstraintTest {
         }
     }
 
+    @Test
+    fun `check against security level - secure enclave`() {
+        val secureEnclave = TestEnclaveInstanceInfo(summary = SECURE)
+
+        listOf(STALE, SECURE).forEach { securityLevelRequiredByClient ->
+            val constraint = EnclaveConstraint.parse("SEC:$securityLevelRequiredByClient C:${secureEnclave.enclaveInfo.codeHash}")
+            assertDoesNotThrow { constraint.check(secureEnclave) }
+        }
+
+        assertThrows<InvalidEnclaveException> {
+            val securityLevelRequiredByClient = INSECURE
+            val constraint = EnclaveConstraint.parse("SEC:$securityLevelRequiredByClient C:${secureEnclave.enclaveInfo.codeHash}")
+            constraint.check(secureEnclave)
+        }
+    }
```

Comment:
Shouldn't parse throw an exception if securityLevelRequiredByClient is STALE?
My understanding is that enclave security level = SECURE => if securityLevelRequiredByClient in {INSECURE, STALE} throws an exception. 
It shouldn't throw an exception only if securityLevelRequiredByClient = SECURE

Reply:
I am not sure if I am understanding your comment. If an enclave is classified as secure then a client that sets the constraint as `STALE` or `SECURE` should be able to connect to that enclave. The client sets the minimum security that it expects the enclave to have. However, if the client wants the min. security level to be insecure then the enclave must be classified as insecure.

Reply:
Nope, that’s not correct. It goes against your PR comment which is correct. 
“If the security level is set to STALE then the client can communicate with a STALE or SECURE enclave. If the security level is set to SECURE then the client can only communicate with a SECURE enclave. “

The security levels are as follows INSECURE <= STALE <= SECURE

Reply:
_“If the security level is set to STALE then the client can communicate with a STALE or SECURE enclave. If the security level is set to SECURE then the client can only communicate with a SECURE enclave. “_
This is what the unit-tests are doing.

Reply:
Oh sorry, I have mixed and matched the security level of an enclave and client's required security level! Looks good now!